torchbearer/callbacks/torch_scheduler.py

```diff
@@ -21,15 +21,22 @@ def on_sample(self, state):
 
     def on_step_training(self, state):
         if self._step_on_batch and self._monitor is not None:
-            self._scheduler.step(state[torchbearer.METRICS][self._monitor])
+            if self._monitor in state[torchbearer.METRICS]:
+                self._scheduler.step(state[torchbearer.METRICS][self._monitor])
 
     def on_start_training(self, state):
         if not self._step_on_batch and self._monitor is None:
             self._scheduler.step(epoch=state[torchbearer.EPOCH])
+        else:
+            warnings.warn(f"Monitor key [{self._monitor}] was not found by the scheduler.", Warning)
```

Comment:
- I think this warning should go after line 25, at the moment it's just emitting a warning whenever we're not stepping on batch. 
- We should probably also use string formatting that works on both python 2 and 3 for the moment, even if python 2 is unsupported these days. 
I believe ``"Monitor key [{}] was not found by the scheduler.".format(self._monitor)`` works on both versions.